src/simplewallet/simplewallet.cpp

```diff
@@ -2504,16 +2504,25 @@ bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::stri
   size_t fake_outs_count = 0;
   if(local_args.size() > 0) {
     size_t ring_size;
-    if(!epee::string_tools::get_xtype_from_string(ring_size, local_args[0]))
+    float ring_size_faulty;
+    if (epee::string_tools::get_xtype_from_string(ring_size, local_args[0]))
     {
-      fake_outs_count = m_wallet->default_mixin();
-      if (fake_outs_count == 0)
-        fake_outs_count = DEFAULT_MIX;
+      // string is an integer: it is the ring size
+      fake_outs_count = ring_size - 1;
+      local_args.erase(local_args.begin());
+    }
+    else if(epee::string_tools::get_xtype_from_string(ring_size_faulty, local_args[0]))
+    {
+      // string is a number but not an integer: failed attempt to specify ring size. Confusion with amount?
+      fail_msg_writer() << tr("ring size must be an integer >= 3: ") << ring_size_faulty;
```

Comment:
This is dependent on version (and currently >= 5).

Reply:
Ah yes, I lazily copy\pasted that string from other errors. These strings for different errors need to be updated too then.